tests/unit/model/projections/changed-attrs-test.js

```diff
@@ -88,4 +88,191 @@ module('unit/model/projections/changed-attrs', function(hooks) {
       'changed attributes is correct'
     );
   });
+
+  test('Can set a many embedded property to a semi resolved array containing a mix of pojos and megamorphic models', function(assert) {
+    assert.expect(3);
+    this.owner.register(
+      'service:m3-schema',
+      class TestSchema extends DefaultSchema {
+        includesModel() {
+          return true;
+        }
+
+        computeNestedModel(key, value) {
+          assert.ok(true, 'Called computed nested model');
+          let attributesType;
+          try {
+            attributesType = value && value.$type;
+          } catch (e) {
+            assert.ok(false, "We don't pass Megamorphic Models to computeNestedModel");
+          }
```

Comment:
what's the point of `try/catch` -> assert.  We'll get a failure if we let this exception go through no?

Reply:
fixed